### PR TITLE
Sc 20594

### PR DESCRIPTION
**Pull request by daniellemaxwell**
Scope of Changes:
Replaces `category` with `tags` in the frontmatter for each blog post file to create a basic `/tags` page for each tag listed on the blog using Hugo's default taxonomy. The `/tags/tag-name` page currently displays each blog post associated with the tag. 

The following changes will be applied in a separate PR:
- Add clickable link to blog post tags
- Add styles to `/tags` page to be more presentable 
- Add header text to the `/tags/tag-name` pages 
- Update blog archetype to replace `category` with `tags`

Fixes SC-20594

Acceptance Criteria:
https://www.awesomescreenshot.com/video/20744405?key=6702142a93dff2931bd8990c11b89290

### CODE REVIEW

File: content/en/blog/2021-01-21-hello-world.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ date: "2021-01-21T16:20:18-04:00"
 draft: false
 image: img/blog/2021-01-21-hello-world/gulfatnight.jpg
 author: Benjamin Bengfort
-category: "Distributed Systems"
+tags: ['Distributed Systems']
```
Comment (daniellemaxwell) [inline content/en/blog/2021-01-21-hello-world.md:8]: Hugo has a built in support for taxonomies to group content. Tags and categories are defined by default. In order for it to work, we have to make the taxonomy plural in the frontmatter. We can create a custom taxonomies, so we could make this `blog-tags` if we want.

I thought `/tags/tag-name` would be shorter, so opted for that over `categories/category-name`. Here's a link to more info about taxonomies: https://gohugo.io/content-management/taxonomies/#default-taxonomies

File: content/en/blog/2021-03-19-a-parrot-trainer-eats-crow.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ date: "2021-03-19T08:42:21-04:00"
 draft: false
 image: img/blog/2021-03-19-a-parrot-trainer-eats-crow/parrots.jpg
 author: Rebecca Bilbro
-category: AI/ML
+tags: ['AI/ML']
```
Comment (daniellemaxwell) [inline content/en/blog/2021-03-19-a-parrot-trainer-eats-crow.md:8]: The page for this tag will appear as `/tags/ai/ml` in the browser. Should we change this to `AI-ML` or separate list it as `['AI', 'ML']`?

File: layouts/_default/article.html [hunk 3]
```diff
@@ -4,7 +4,7 @@
     <div class="bg-[#ECF6FF] rounded-b-xl ">
 
         <div class="px-6 pt-4">
-            <h4 class="text-base font-bold text-[#1D65A6]">{{ .Params.category }}</h4>
+            <h4 class="text-base font-bold text-[#1D65A6]">{{ delimit .Params.tags ", " }}</h4>
```
Comment (daniellemaxwell) [inline layouts/_default/article.html:7]: Adding `delimit` here ensures that the tags don't appear exactly as they look in the frontmatter.

File: layouts/_default/list.html [hunk 4]
```diff
@@ -1,9 +1,7 @@
-{{ define "main" }}
-  <h1>Posts</h1>
-  {{ range .Pages }}
-    <article>
-      <h2>{{ .Title }}</h2>
-      {{ .Content }}
-    </article>
+<ul>
```
Comment (daniellemaxwell) [inline layouts/_default/list.html:1]: This is a basic style add to display each tag in a list and will be updated in a separate PR.

File: content/en/blog/2021-03-19-a-parrot-trainer-eats-crow.md (hunk 2 above)
Comment (rebeccabilbro) [reply content/en/blog/2021-03-19-a-parrot-trainer-eats-crow.md:8]: Great point - yes, let's split into 2 tags: ['AI', 'ML']

**Review by rebeccabilbro — APPROVED** [review]
Thank you @daniellemaxwell for doing the research to figure out how to do this the "hugo way" -- this looks like a great start and I look forward to seeing how the stylized tags page looks. One minor thing I noticed is that the `/tags/tag-name` pages includes posts that are not officially published yet (e.g. where `draft: true` in the markdown frontmatter metadata), like this one:

Screenshot 2023-09-14 at 10 16 18 AM

Is there an easy way to exclude those draft posts from showing up in the `tags` page? This shouldn't block merging, since there are only one or two posts still in draft form.

**Comment by daniellemaxwell** [comment]
@rebeccabilbro Great catch! I think that adding a condition to check for the post's draft status should resolve that issue.
